Automattic/jetpack
# Implement VR shortcode in Gutenberg

Adds Gutenberg support for the `[vr]` shortcode.

#### Changes proposed in this Pull Request:

* Add gutenberg block for `[vr]` shortcode.
* Adds some basic infrastructure to support future Gutenberg blocks:
  * Inject locale info when rendering Gutenberg editor.
  * .gitignore generated block files

#### Known issues:

* Doesn't allow changing view type or image after creating
* Doesn't allow drag and drop of image onto control

#### Testing instructions:

* Install Gutenberg
* Run `gulp gutenpack` in this Jetpack branch
* Edit a post in Gutenberg
* Add a "VR Image" block to any post
* Paste in the URL of any 360' image, e.g. https://en-blog.files.wordpress.com/2016/12/regents_park.jpg
* Save the post

Your post should now have a 360' photo element embedded in it!

Switching between "360" and "Cinema" changes how the renderer treats the image - as a spherical image (360) or a panorama (cinema).


#### Proposed changelog entry for your changes:

Add Gutenberg support for embedding VR photos.

#### Screenshot

![goldsounds ngrok io_2018_05_15_my-vr-post_ ipad](https://user-images.githubusercontent.com/51896/40092623-03a9c6ea-5873-11e8-8b87-8aa07e19b283.png)
